### ISSUE TITLE
Question on Dataset Prepare

### ISSUE THREAD

**Issue by UniGen-X** [ShuangLI59/unified_video_action#6]
Hi,

This is a really excellent work. I would like to try it on my own robot data. I guess the most easy way is to change my own dataset to be a UMI dataset format, right? 

My current dataset is under lerobot format, I was wondering if you knew any existing scripts to directly transfer between these two dataset formats?

Thanks!

**Comment by ShuangLI59** [comment]
Hi @robot-gen , thank you for your interest!

Yes, the data format can be similar to UMI. However, there might be some modifications depending on your specific needs. The UMI dataset uses a wrist-mounted camera with a fisheye view, so we predict relative actions, which are then converted to absolute actions during deployment. In the UMI experiment, we did not attempt to directly predict absolute actions. However, we have successfully predicted absolute actions in simulation experiments, such as those with Libero and Robomimic. For your task, please choose the appropriate action representation (relative or absolute) based on your requirements.

Also, for real-world experiments with larger datasets, we recommend using a larger model size (`mar_large`, or `mar_huge`). The current default model is relatively small. You can increase the model size by modifying the command below, and downloading the pretrained weights from the following link:
https://github.com/LTH14/mar

Place the downloaded weights under the pretrained_models folder, and set the model parameters as follows:
```
model.policy.autoregressive_model_params.pretrained_model_path="pretrained_models/mar/mar_huge/checkpoint-last.pth" \
model.policy.autoregressive_model_params.model_size=mar_huge \
```

**Comment by UniGen-X** [comment]
Thank you so much for the quick reply and all details above!

**Comment by robot-gen** [comment]
Hi @ShuangLI59 ,

Thank you so much again for the above notes! I have carefully read into you code and gotten promising progress. However, there were still several things I would like to double-check with you. I would appreciate if you could help me with these:

1. why do you downsample 3 steps for umi-task particularly, but not for Libro and Pusht tasks? 
2. The above downsample process also takes to actions, then the action frequency output by the model would 3 times lower than the origin control frequency in the data, right?
3. For any other real-world tasks do you suggest starting with downsample disabled, or 3 steps?
4. Although you have the following setup in [umi_multi.yaml](https://github.com/ShuangLI59/unified_video_action/blob/5412ca5ababfef40ef15b23ca2d59c0be59d3810/unified_video_action/config/task/umi_multi.yaml#L39),

```
low_dim_obs_horizon: 32
img_obs_horizon: 32
action_horizon: 32
```
the `horizon of obs history` is actually `4`, and predicting future `4 `frames and future `16` actions, where the ```history of obs horizon 4``` is achieved in the line 339 in [data_utils.py](https://github.com/ShuangLI59/unified_video_action/blob/5412ca5ababfef40ef15b23ca2d59c0be59d3810/unified_video_action/utils/data_utils.py#L338), right?
5. For the horizon of observation history, do you have any specific suggestions for any other real-world tasks? I feel like people usually choose 2 or 4.

Training question: 
6. You mentioned that each stage training takes 2 days with 8xH100. In general, do you have any suggestion about how many epochs your method needed to fine-tuning on a new real-world tasks? Or the scale of the loss value before stopping?

For the following two questions, I may miss something in your code:
7. In the function ```__getitem__``` of ```umi_lazy_dataset.py``` [here](https://github.com/ShuangLI59/unified_video_action/blob/5412ca5ababfef40ef15b23ca2d59c0be59d3810/unified_video_action/dataset/umi_lazy_dataset.py#L273), the image are selected by using a ```include_indices``` that are different with the one for other proprioception data, then, how do you guarantee the proprioception info correctly aligned with the images on timesteps in observation?
8. I am not quite familiar with UMI tasks. It seems that you were not doing normalization for proprioception and action data. Is this because, the data has already normalized in the original dataset?

Look forward to hearing from you soon, and I appreciate!!!!

**Comment by UniGen-X** [comment]
Hi,

Sorry for the previous huge questions! After taking a deeper look into your code, I have found some answers. I just would like to further check:

1- It seems that, during training, the video frames are selected by a stride ```4```, any particular reason to do this? 
2- It seems that, no matter "video_model" or "policy_model", all latent z are used as condition for future frame-generation and action generation. However, Fig. 2 shows "each latent token" for video-generation, but "all latent tokens" for action-generation. Did I misunderstand anything?
3- Do you have any suggestion about how many epochs your method needed to fine-tuning on a new real-world tasks? Or the scale of the loss value would be before stopping?

Thank you very much, and look forward to hearing from you!

**Comment by ShuangLI59** [comment]
HI @robot-gen thank you for your questions:

1: We selected 16 actions and used a lower frame rate for the videos due to the high redundancy in the visual information. The stride of 4 was chosen empirically, but feel free to experiment with different stride values to see what works best.

2: We learn a unified latent representation 𝑧, which can be used for video generation, action generation, or both—depending on which diffusion head is used.

3: It's difficult to estimate without knowing the specific task and dataset. For UMI tasks, we trained the model for around 40 epochs.

**Comment by UniGen-X** [comment]
Thank you so much for the response! It helps a lot.
